### ISSUE TITLE
Sigmac converting rules to stix throws error "Backend does not support map values of type <class 'sigma.parser.condition.NodeSubexpression'>"

### ISSUE THREAD

**Issue by ali-saad-jaffer** [SigmaHQ/sigma#3486]
A lot of rules are now returning the error that `Backend does not support map values of type <class 'sigma.parser.condition.NodeSubexpression'>` when they are converted to Stix.

I am using the following command to convert the rules
```
python3 tools/sigmac --output-format json --config stix-shifter --config stix2.0 --target stix rules/windows/builtin/system/win_invoke_obfuscation_via_use_rundll32_services.yml
```

I noticed that this error started to happen after this commit(https://github.com/SigmaHQ/sigma/commit/1fbd2bba4dcddfa0223a1d34dea893be6d3f9913). Can this be fixed in sigmac or some config update that can be done to make it work?

The error is being thrown for a lot of rules, here is a list of few of them :
https://github.com/SigmaHQ/sigma/blob/master/rules/windows/builtin/system/win_invoke_obfuscation_via_use_rundll32_services.yml
https://github.com/SigmaHQ/sigma/blob/master/rules/windows/network_connection/net_connection_win_remote_powershell_session_network.yml
https://github.com/SigmaHQ/sigma/blob/master/rules/windows/builtin/system/win_invoke_obfuscation_via_rundll_services.yml
https://github.com/SigmaHQ/sigma/blob/master/rules/windows/builtin/security/win_invoke_obfuscation_via_compress_services_security.yml
https://github.com/SigmaHQ/sigma/blob/master/rules/windows/builtin/system/win_invoke_obfuscation_clip_services.yml

**Comment by thomaspatzke** [comment]
Ok, seems the fix from yesterday introduced some other bug. Thanks for pointing this out!

**Comment by thomaspatzke** [comment]
No bug as I initially thought...seems that older *generateMapItemNode* method implementations without support for *NodeSubexpression* were copied over to backend code instead of using inheritance. This code is the total mess.

**Comment by thomaspatzke** [comment]
Fixed, check release 0.22.1 or the current master branch.
